PWG/DevNanoAOD/AliAnalysisNanoAODCuts.cxx

```diff
@@ -734,6 +736,44 @@ Bool_t AliAnalysisNanoAODEventCuts::IsSelected(TObject* obj)
   return kTRUE;
 }
 
+AliAnalysisNanoAODMCParticleCuts::AliAnalysisNanoAODMCParticleCuts()
+    : AliAnalysisCuts(),
+      fDoSelectPrimaries(true),
+      fDoSelectCharged(true),
+      fForceKeepParticles(false),
+      fMinPt(0.f),
+      fMaxEta(999.f),
+      fPDGToKeep(),
+      fPDGV0() {
+}
+
+bool AliAnalysisNanoAODMCParticleCuts::IsSelected(TObject* obj) {
+  // Returns true if the MC particle is to be kept!
+  AliAODMCParticle* part = dynamic_cast<AliAODMCParticle*>(obj);
+
+  // if this is activated the specified particles are kept ignoring the cuts
+  if (fForceKeepParticles) {
```

Comment:
Why is this needed? For performance? If the list is empty (default), it will just continue...